docs/workflow.md

```diff
@@ -26,6 +34,23 @@ Draft pull requests will be closed within 30 days.
 
 The active development branch in our repo is `dev`. What we ship comes from the `release-major.minor.x` branches.
 
+#### Adding fixes in release branches
+
+NuGet primarily works on the dev branch and that's where most of the commmits will be merged. At a certain point, NuGet branches to a release branch during the release stabilization phase and the last few commits usually go into that branch.
+
+In most cases, a fix will be originally developed on the dev branch and then moved to the release branch.
+When moving to the release branch the recommended approach is the following:
+
+- Create a new branch based on the release branch.
+- Cherry pick the commit of interest
+- Push
+- Create a Pull Request against the release branch.
+
+The cherry-picked request does not need to be reviewed, but when the build passes it can be merged.
+The recommended pattern for release branches is slightly different. We should preserve the original commit message one can follow the original Pull Request.
+
+![Good Release branch Commit Message](resources/release-branch-commit-message.png)
```

Comment:
You're usually going to cherry-pick 1 by 1. 
That way in case something does go wrong, only that commit will be reverted. I'll add details about that.